include/posix/pthread.h

```diff
@@ -7,38 +7,57 @@
 #ifndef __PTHREAD_H__
 #define __PTHREAD_H__
 
-#ifdef CONFIG_NEWLIB_LIBC
+#include <kernel.h>
 #include <time.h>
-#else
-/* This should probably live somewhere else but Zephyr doesn't
- * currently have a stdc layer to provide it
- */
-struct timespec {
-	s32_t tv_sec;
-	s32_t tv_nsec;
+#include "sys/types.h"
+#include "posix_sched.h"
+#include "unistd.h"
+
+enum pthread_state {
+	/* The thread is running and joinable. */
+	PTHREAD_JOINABLE = 0,
+	/* The thread is running and detached. */
+	PTHREAD_DETACHED,
+	/* A joinable thread exited and its return code is available. */
+	PTHREAD_EXITED,
+	/* The thread structure is unallocated and available for reuse. */
+	PTHREAD_TERMINATED
 };
-#endif
 
-static inline s32_t _ts_to_ms(const struct timespec *to)
-{
-	return (to->tv_sec * 1000) + (to->tv_nsec / 1000000);
-}
+struct posix_thread {
+	struct k_thread thread;
+
+	/* Exit status */
+	void *retval;
 
-typedef struct pthread_mutex {
-	struct k_sem *sem;
-} pthread_mutex_t;
+	/* Pthread cancellation */
+	int cancel_state;
+	int cancel_pending;
+	struct k_sem cancel_lock_sem;
+	pthread_mutex_t cancel_lock;
+
+	/* Pthread State */
+	enum pthread_state state;
+	pthread_mutex_t state_lock;
+	struct k_sem state_lock_sem;
+	pthread_cond_t state_cond;
+};
 
-typedef struct pthread_mutexattr {
-	int unused;
-} pthread_mutexattr_t;
+/* Pthread detach/joinable */
+#define PTHREAD_CREATE_JOINABLE     0
+#define PTHREAD_CREATE_DETACHED     1
 
-typedef struct pthread_cond {
-	_wait_q_t wait_q;
-} pthread_cond_t;
+/* Pthread cancellation */
+#define PTHREAD_CANCEL_POS	0
```

Comment:
Any #define that is an implementation detail should not be exported. Move these to the .c file, or use another namespace for them (e.g. prefix with _).

Reply:
Sure I will modify this,